Add user interface for graph editor graph layout parameters (#406)
hexatomic/hexatomic: docs/user/src/usage/graph-editor/layout-params.md

# Change layout parameters

> Depending on the shown annotations and the specific graph, the default visualization of the graph might get cluttered.

Reviewer: ```suggestion
Depending on the visible annotations and the specific graph, the default visualization of the graph might be cluttered.
```

Improve wording.

> The graph editor will try to arrange the nodes using a default layout algorithm, which parameters you can adjust by expanding the "Graph Layout" section.

Reviewer: ```suggestion
The graph editor will try to arrange the nodes using a default layout algorithm.
You can adjust the layout parameters by expanding the "Graph Layout" section.
```

Separate sentences.

![Graph editor view with expanded layout parameters](./layout-params.png)

> ## Horizontal margin between token

Reviewer: ```suggestion
## Horizontal margin between tokens
```

Use plural

![Horizontal margin parameter slider](./layout-param-horizontal-margin.png)

> There is some horizontal space between the token at the bottom of the graph.

Reviewer: ```suggestion
This parameter changes the horizontal space between the tokens at the bottom of the graph.
```

Improve wording.

If the space is increased, this indirectly affects the space between other nodes, too.

> This is measured in "times of the average token width".

Reviewer: ```suggestion
This parameter is measured in "multiples of the average token width".
```

Improve wording.

> So for \"0\" there is no margin, for "1" the margin has  the same width as the average token node, and for "2" the margin is twice as high as the average token node width.

Reviewer: ```suggestion
So for \"0\" there is no margin, for "1" the margin has the same width as the average token node, and for "2" the margin is twice as high as the average token node width.
```

Remove duplicate whitespace.

![Effect of the settings 0, 1 and 2 for the horizontal margin parameter](horizontal-margin-examples.png)

## Vertical margin between nodes

![Vertical margin parameter slider](./layout-param-vertical-margin.png)

> Configures the vertical margin between *all* nodes.

Reviewer: ```suggestion
This parameter configures the vertical margin between *all* nodes.
```

Make this a sentence.

> This is measure in "times of the node height".

Reviewer: ```suggestion
This is measured in "multiples of the node height".
```

Fix typo, wording.

So for "0" there is no margin, for "1" the margin has the same height as the node.

![Effect of the settings 0, 1 and 2 for the vertical margin parameter](vertical-margin-examples.png)

> ## Vertical margin between token and non-token

Reviewer: ```suggestion
## Vertical margin between tokens and non-tokens
```

Plural.

![Vertical margin between token and non-token parameter slider](./layout-param-token-margin.png)

> Token are grouped in the lowest rank (at the bottom)

Reviewer: ```suggestion
Tokens are grouped horizontally at the bottom of the graph visualization, i.e., in the bottom "row".
```

Introduce row before it is used later, improve wording.

> To allow space for pointing relations, you can add a margin between the token row and the lowest annotation nodes.

Reviewer: ```suggestion
To allow space for pointing relations, you can add a margin between the token "row" and the annotation nodes in the "row" just above it.
```

This margin is in addition to the [vertical margin between nodes](#vertical-margin-between-nodes).

> A margin of "1" means there is one level left empty, an offset of "0"

Reviewer: ```suggestion
A margin of "1" means there is one empty level added between the token "row" and the annotation node "row" above it, an offset of "0"
```

Make clearer.
